src/clarity-angular/datagrid/datagrid-hidable-column.directive.ts

```diff
@@ -0,0 +1,121 @@
+import {
+    Directive,
+    TemplateRef,
+    ViewContainerRef,
+    Input,
+    AfterViewInit
+} from "@angular/core";
+
+import { DatagridHideableColumn } from "./datagrid-hideable-column";
+import { DatagridColumn } from "./datagrid-column";
+
+@Directive({
+    selector: "[clrDgHideableColumn]"
+})
+
+/**
+ * @class DatagridHideableColumnDirective
+ *
+ * @description
+ * A structural directive meant to be used inside a clr-dg-column component.
+ *
+ * <clr-dg-column>
+ *       <ng-container *clrDgHideableColumn="{ hidden: true }">
+ *           User ID
+ *       </ng-container>
+ *   </clr-dg-column>
+ *
+ * It sets up state and properties so that columns can be manges for hide/show by a service and an internal
+ * datagrid toggle component.
+ *
+ */
+export class DatagridHideableColumnDirective implements AfterViewInit {
+
+    /**
+     * @property _hidden
+     *
+     * @description
+     * Used to initialize the column with either hidden or visible state.
+     *
+     * @type boolean
+     */
+    private _hidden: boolean;
+
+    /**
+     * @function clrDgHideableColumn
+     *
+     * @description
+     * Setter fn for the @Input with the same name as this structural directive.
+     * It allows the user to pre-configure the column's hide/show state. { hidden: true }
+     * It's more verbose but has more Clarity.
+     *
+     * @default false
+     *
+     * @type object
+     *
+     * @example
+     * *clrDgHideableColumn
+     * *clrDgHideableColumn={hidden: false}
+     * *clrDgHideableColumn={hidden: true}
+     *
+     * @param value
+     *
+     */
+    @Input("clrDgHideableColumn")
+    set clrDgHideableColumn( value: any ) {
+        this._hidden = (value) ? value.hidden : false;
```

Comment:
I am not sure how a default (`false`) could tackle the error that could arise when a user passes an object with a property that has a different name than `hidden`. But do we have to ask an object as a parameter here instead of asking just a boolean value so we could avoid all these checkings?  If the value parameter has to be an object, we could also use a simple interface on the value parameter?